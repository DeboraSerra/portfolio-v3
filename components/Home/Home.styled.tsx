import styled from "styled-components";

export const Main = styled.div`
  min-height: 74vh;
  background-color: ${props => props.theme.bg.primary};
  padding: 12px 24px;
`;
