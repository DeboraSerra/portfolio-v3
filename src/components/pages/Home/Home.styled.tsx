import styled from "styled-components";

export const Main = styled.main`
  padding: 40px 0;
  background-color: ${props => props.theme.bg.tertiary};

  .home {
    &__intro {
      margin: 24px 0;
      padding-top: 28px;
      border-top: 1px solid ${props => props.theme.bg.contrast[3]};

      & img {
        width: 500px;
        height: 500px;
        object-fit: cover;
        object-position: 50% 60%;
        border-radius: 50%;
      }// & img

      &--right {
        min-width: 50%;

        text-align: center;
      }// &--right

      @media only screen and (max-width : 992px) {
        flex-direction: column;
        gap: 48px;

        & img {
          max-width: 100%;
          height: auto;
          aspect-ratio: 1/1;
        }// & img
      }// @media only screen and (max-width : 992px)
    }// &__intro
  }// .home
`;
