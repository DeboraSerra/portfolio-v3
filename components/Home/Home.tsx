import { NextPage } from "next";
import Image from "next/image";
import Profile from "../../assets/images/home/profile.jpg";

import * as S from "./Home.styled";

interface Props {}

const Home: NextPage<Props> = () => {
  return (
    <S.Main>
      <h1 className="home__title">Hello world!</h1>
      <h2 className="home__subtitle">My name is Débora Serra</h2>
      <Image
        src={Profile}
        alt="Débora Serra"
        width={500}
        height={600}
        className="home__img"
      />
      <p className="home__description">
        Web developer student in Trybe | Masters degree in Chemistry by UnB
      </p>
      <p className="home__description">
        Cosmetologist by iCosmetologia | Post-graduated in Gastronomy in PUC-RS
      </p>
      <p className="home__description">
        See my{" "}
        <a
          href="https://github.com/DeboraSerra"
          target="_blank"
          rel="noopener noreferrer"
        >
          GitHub
        </a>
      </p>
    </S.Main>
  );
};

export default Home;
