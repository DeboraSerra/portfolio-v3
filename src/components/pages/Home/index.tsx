import { Subtitle, Text, Title } from "@/styles/styled";
import * as S from "./Home.styled";

import profile from "../../../../assets/images/home/me.jpg";
import Image from "next/image";
import Timeline from "@/components/Timeline";

const Home = () => {
  return (
    <S.Main>
      <div className='container'>
        <Title className='center'>Welcome to my Portfolio!</Title>
        <div className='home__intro flex center'>
          <Image src={profile} alt='Débora Serra' width={388.0} height={518.4} />
          <div className='home__intro--right'>
            <Subtitle>Débora Serra</Subtitle>
            <Text>Full Stack Web Developer | Chemist</Text>
            <Text>Baker | Cosmetologist</Text>
          </div>
        </div>
        <Timeline />
      </div>
    </S.Main>
  );
};

export default Home;
