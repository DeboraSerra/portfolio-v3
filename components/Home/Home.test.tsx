import { render } from "@testing-library/react";
import Home from "./Home";
import TProvider from "../helpers/ThemeProvider";

describe("Test the Home component", () => {
  it('Should match the snapshot', () => {
    const { asFragment } = render(<TProvider><Home /></TProvider>)
    expect(asFragment()).toMatchSnapshot();
  })
})