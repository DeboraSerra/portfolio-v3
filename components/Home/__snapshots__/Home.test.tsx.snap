// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Home component Should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  min-height: 74vh;
  background-color: #e8ccc7;
  padding: 12px 24px;
}

<div
    class="c0"
  >
    <h1
      class="home__title"
    >
      Hello world!
    </h1>
    <h2
      class="home__subtitle"
    >
      My name is Débora Serra
    </h2>
    <img
      alt="Débora Serra"
      class="home__img"
      data-nimg="1"
      decoding="async"
      height="600"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=1080&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1080&q=75 2x"
      style="color: transparent;"
      width="500"
    />
    <p
      class="home__description"
    >
      Web developer student in Trybe | Masters degree in Chemistry by UnB
    </p>
    <p
      class="home__description"
    >
      Cosmetologist by iCosmetologia | Post-graduated in Gastronomy in PUC-RS
    </p>
    <p
      class="home__description"
    >
      See my 
      <a
        href="https://github.com/DeboraSerra"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </p>
  </div>
</DocumentFragment>
`;
